File: 240311/스승의 은혜 2/the-grace-form-teacher-2.py
n,b=map(int,input().split())
price=[int(input()) for _ in range(n)]

ans=0

for i in range(n):
    cnt_price=0
    num=0
    for j in range(n):
        if i==j:
            if cnt_price+price[j]//2<=b:
                cnt_price+=price[j]//2
                num+=1
            else:
                break
        else:
            if cnt_price+price[j]<=b:
                cnt_price+=price[j]
                num+=1
            else:
                continue

    ans=max(ans,num)

print(ans)

File: 260118/스승의 은혜 2/the-grace-form-teacher-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, b] = input[0].split(' ').map(Number);
const p = input.slice(1, 1 + n).map(Number);

p.sort((a,b)=> a-b);

let ans = 0;

for(let i=0; i<n; i++){
    let cnt=0, cost=0;
    for(let j=0; j<n; j++){
        if(i===j){
            cost+=p[j]/2;
        }else{
            cost+=p[j];
        }
        if(cost<=b){
            cnt++;
        }else{
            break;
        }
    }
    ans = Math.max(ans, cnt)
}

console.log(ans)